cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -476,6 +476,8 @@ std::shared_ptr<RecordBatchReader> MakeGeneratorReader(
       return Status::OK();
     }
 
+    Status Close() override { return Status::OK(); }
```

Comment:
This should read from the generator until the end is reached.

Reply:
👍